### PR TITLE
[video_player_avfoundation] Support the audio-only HLS (.m3u8) on iOS

### PR DESCRIPTION
**Pull request by napoleon-na**
Supports the audio-only HLS on iOS. 

After through https://github.com/flutter/plugins/pull/4639 and https://github.com/flutter/plugins/pull/4727, we can play HLS videos and audios, but it does not support audio-only HLS. When attempting to play that on iOS, it never finishes initialization.

Show the before/after demonstration here: https://github.com/flutter/flutter/issues/156589#issuecomment-2408846446

Fixes: https://github.com/flutter/flutter/issues/156589
Test with https://github.com/flutter/assets-for-api-docs/pull/252

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

### CODE REVIEW

**Comment by napoleon-na** [comment]
I would appreciate checking https://github.com/flutter/assets-for-api-docs/pull/252 first to pass the tests in this PR. Although I read the docs, let me know if I need to go through a different process.

File: packages/video_player/video_player_avfoundation/CHANGELOG.md
Comment (cbracken) [inline packages/video_player/video_player_avfoundation/CHANGELOG.md:3]:
```suggestion
* Bugfix to allow audio-only HLS (.m3u8) on iOS.
```

**Review by cbracken — COMMENTED** [review]
lgtm once asset patch has been verified and landed.

File: packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m
Comment (hellohuanlin) [inline packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m:434]: can you explain why this is related to video vs audio tracks? (not quite sure how this fixed the audio only files)

Comment (napoleon-na) [reply packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m:434]: @hellohuanlin 
This is because HLS files AVAsset does not always have the track information because HLS is a streaming file, so we're checking `currentItem.tracks.count` to solve this problem.

Also, it is related to the size. HLS video files have height and width, but the HLS audio files do not. The differences affect the later if-statement, and that's why adding the checking above.

File: packages/video_player/video_player_avfoundation/CHANGELOG.md
Comment (napoleon-na) [reply packages/video_player/video_player_avfoundation/CHANGELOG.md]: @cbracken 
addressed this in the following commit!
[Tweak the CHANGELOG](https://github.com/flutter/packages/pull/7890/commits/041189a82881fdb78fffb249a7da47d1ae3c78b9)

File: packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m
Comment (hellohuanlin) [reply packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m:454]: So for audio only files, `currentItem.tracks` is not empty, but asset.tracks is empty?

Comment (napoleon-na) [reply packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m:454]: Yes, that's right! 🙂

Comment (hellohuanlin) [reply packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m:454]: This kind of stuff needs to be commented since it's not apparent at first. Could you add some comment? Thank you!

Comment (napoleon-na) [reply packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m:456]: Added some comments in the below commit. Tell me a better comment if you have:
[Add clarification comment for checking HLS audio-only tracks](https://github.com/flutter/packages/pull/7890/commits/ac6c52859092a5174eff60c1bb7e2579c64f06db)

File: packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m
Comment (hellohuanlin) [inline packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m:455]: This doesn't explain why for audio only HLS, track info must be checked using `currentItem.tracks`. What's special about audio only HLS?

Comment (napoleon-na) [reply packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m]: @hellohuanlin 
Thank you for your feedback! I improved the comment to clear the reason (due to the format and no size):
[Improve the comment for checking HLS audio-only tracks](https://github.com/flutter/packages/pull/7890/commits/fae1d41ddc74af2ef2a9a89edb75f5f99539a77a)

**Review by hellohuanlin — APPROVED** [review]

**Comment by napoleon-na** [comment]
@cbracken @hellohuanlin 
All checks have passed after the latest updates. Please let me know if there are any further changes. Thank you!

**Review by cbracken — APPROVED** [review]

**Comment by nateshmbhat** [comment]
when is this getting published as a new video_player version  ?

**Comment by hellohuanlin** [comment]
@nateshmbhat it should already be published

**Comment by nateshmbhat** [comment]
I'm still seeing video_player library in pub dev showing last update was 3 months back

**Comment by hellohuanlin** [comment]
> I'm still seeing video_player library in pub dev showing last update was 3 months back

It's the implementation library: https://pub.dev/packages/video_player_avfoundation

**Comment by nateshmbhat** [comment]
So if I use the current version of video player will it automatically use this new AV Foundation Player version?

**Pull request merged**
